contracts/AtlasMineStakerUpgradeable.sol

```diff
@@ -983,6 +1002,36 @@ contract AtlasMineStakerUpgradeable is
         return staked + _totalUsableMagic();
     }
 
+    /**
+     * @dev Determine if the current block timestamp falls in an accrual window.
+     *      During accrual windows deposits/withdraws/claims are enabled, and reward
+     *      updating is enabled.
+     *
+     * @return inWindow             Whether the current time falls in an accrual window.
+     */
+    function _isAccrualWindow() internal view returns (bool inWindow) {
+        require(accrualWindows.length > 0, "Accrual windows not set");
+        /// time elapsed in day / hours
+        uint256 currentHour = (block.timestamp % 86_400) / 3_600;
+
+        for (uint256 i = 0; i < accrualWindows.length - 1; i += 2) {
+            if (currentHour >= accrualWindows[i] && currentHour < accrualWindows[i + 1]) {
+                inWindow = true;
+                break;
+            }
+        }
+    }
```

Comment:
do we have to set `inWindow = false` at the end?

Reply:
The default value for bool is false so it's okay not to add `inWindow = false`. But if you want to add the line for explicit, the line must be added in front of for loop.

Reply:
What minebuu said, style-wise, I prefer to not add it, and leverage the default